src/main/java/edu/ie3/datamodel/io/csv/BufferedCsvWriter.java

```diff
@@ -73,15 +75,16 @@ public BufferedCsvWriter(String baseFolder, CsvFileDefinition fileDefinition, bo
    */
   public synchronized void write(Map<String, String> entityFieldData)
       throws IOException, SinkException {
-    /* Check against eligible head line elements */
+    /* Check against eligible headline elements */
     if (entityFieldData.size() != headLineElements.length
         || !entityFieldData.keySet().containsAll(Arrays.asList(headLineElements)))
       throw new SinkException(
           "The provided data does not meet the pre-defined head line elements '"
               + String.join(",", headLineElements)
               + "'.");
 
-    String[] entries = entityFieldData.values().toArray(new String[0]);
+    String[] entries =
+        Arrays.stream(headLineElements).map(entityFieldData::get).toArray(String[]::new);
     writeOneLine(entries);
```

Comment:
Here, a stream terminal operation (toArray) is used. You may consider adapting the writeOneLine to take a Stream as input parameter. This would allow for out of the box appending the elements of the stream avoiding the overhead of array allocation. But this is a very minor thing that came into my mind when reading this line ...